tests/ui/components/search/searchcomponent.js

```diff
@@ -0,0 +1,45 @@
+import DOM from '../../../../src/ui/dom/dom';
+import mockManager from '../../../setup/managermocker';
+import { mount } from 'enzyme';
+import GlobalStorage from '../../../../src/core/storage/globalstorage';
+import PersistentStorage from '../../../../src/ui/storage/persistentstorage';
+import StorageKeys from '../../../../src/core/storage/storagekeys';
+
+DOM.setup(document, new DOMParser());
+
+describe('SearchBar component', () => {
+  let COMPONENT_MANAGER;
+
+  const defaultConfig = {
+    container: '#test-component',
+    verticalKey: 'verticalKey'
+  };
+
+  COMPONENT_MANAGER = mockManager({
+    globalStorage: new GlobalStorage(),
+    persistentStorage: new PersistentStorage(),
+    setQuery: jest.fn(),
+    autoCompleteVertical: jest.fn()
+  });
+
+  const globalStorage = COMPONENT_MANAGER.core.globalStorage;
+
+  beforeEach(() => {
+    let bodyEl = DOM.query('body');
+    DOM.empty(bodyEl);
+    DOM.append(bodyEl, DOM.createEl('div', { id: 'test-component' }));
+  });
+
+  it('clicking the clear button deletes SEARCH_OFFSET from global storage', () => {
```

Comment:
I think the clear button does a couple of other things too. Since we're here already, could we make a `clicking the clear button updates storage correctly` test or something like that? It would make sure `SEARCH_OFFSET` is removed from global and persistent storage, `QUERY` is set to `''`, etc.

Reply:
sure, what do you think about making that a describe block and have multiple tests for each of those storage checks. That way we can keep the tests clean with one or two expects per test

Reply:
Sure, that sounds good!